### ISSUE TITLE
tf.math ops do not work on MirroredVariables

### ISSUE THREAD

**Issue by kevin3-black** [tensorflow/tensorflow#31665]
**System information**
- Have I written custom code: yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): macOS Mojave 10.14.5
- TensorFlow installed from (source or binary): binary (tf-nightly)
- TensorFlow version (use command below): 1.15.0-dev20190729
- Python version: 3.7.4

**Describe the current behavior**
When I am using the `tf.distribute.MirroredStrategy` with multiple replicas, `tf.math` ops do not work on `MirroredVariables` when inside a cross-replica scope.

**Describe the expected behavior**
The [MirroredVariable class](https://github.com/tensorflow/tensorflow/blob/e19c354920c3b246dda6598229210a582caaa1a9/tensorflow/python/distribute/values.py#L782) is a subclass of the [DistributedDelegate class](https://github.com/tensorflow/tensorflow/blob/e19c354920c3b246dda6598229210a582caaa1a9/tensorflow/python/distribute/values.py#L375), which if I understand correctly, means that `MirroredVariables` are supposed to act like regular tensors that you can perform ops on. This works fine with most standard ops, like multiplication, division, subtraction, etc. However, if you try to use any `tf.math` ops, you get `TypeError: Failed to convert object of type <class 'tensorflow.python.distribute.values.Mirrored'> to Tensor`.

**Code to reproduce the issue**
```Python
import tensorflow as tf

def merge(strategy, var):
  var = strategy.extended.reduce_to(tf.distribute.ReduceOp.SUM, var, var)
  # return var + 1  # this does work
  return tf.math.add(var, 1)  # this doesn't work

def run(var):
  var = var * var
  return tf.distribute.get_replica_context().merge_call(merge, args=(var,))

strategy = tf.distribute.MirroredStrategy(['/cpu:0', '/cpu:1'])
with strategy.scope():
  var = tf.Variable(2, dtype=tf.float32)
  result = strategy.experimental_run_v2(run, args=(var,))
  print(result)
```

**Comment by oanush** [comment]
@kevin3-black ,
Can you please refer this [link](https://www.tensorflow.org/install/docker).Thanks!

**Comment by kevin3-black** [comment]
Sorry, I forgot to comment out a line. I've updated the post, and reproduced the result on Google Colab: https://colab.research.google.com/drive/1pLeHBPaZxC_OlQw4hXfGyC4gJtZ5e5gT

**Comment by anj-s** [comment]
This issue should be fixed with [commit](1348a03037e6ea79129024f318f8ae9b8f268df8). Please reopen if the issue still persists. Thanks!
